text/0000-introduce-a-new-signing-mechanism.md

```diff
@@ -0,0 +1,568 @@
+[… documentation output …]
+2 gems installed
+```
+
+Some gems are unsigned. If a gem is unsigned, the installation will fail with an error message. For example:
+
+```console
+$ gem install unsigned-foo
+ERROR: `unsigned-foo` is not signed.
+```
+
+This behavior can be suppressed on a per-gem basis. For example:
+
+```console
+$ gem install unsigned-foo unsigned-bar --no-verify-signature unsigned-foo --no-verify-signature unsigned-bar
+Fetching unsigned-foo-1.0.0.gem
+Fetching unsigned-bar-1.0.0.gem
+WARN: unsigned-foo-1.0.0 is unsigned
+WARN: unsigned-bar-1.0.0 is unsigned
+Successfully installed unsigned-foo-1.0.0
+Successfully installed unsigned-bar-1.0.0
+[… documentation output …]
+2 gems installed
+```
+
+### Verifying a gem without installing
+
+Gem signatures can be verified using the `gem signatures --verify GEMNAME` command. For example, to verify the signature of `foo`:
+
+```console
+$ gem fetch foo
+Fetching foo-1.0.0.gem
+Downloaded foo-1.0.0.gem
+
+$ gem signatures --verify foo
+Verified signature(s) for foo-1.0.0.gem
+```
+
+Verifying using `gem signatures --verify`, including verifying multiple gem signatures and the treatment of unsigned gems, is otherwise the same as `gem install`.
+
+## Reference-level explanation
+
+Our discussion at the reference level is assumed to be at Phase 1. This is for two reasons. Firstly, Phase 1 most closely resembles the [proof of concept RubyGems plugin](https://github.com/Shopify/ruby-sigstore) that we have developed from an early sigstore prototype. Secondly, we don’t expect radical changes in implementation for phases 2 and 3, except to improve performance by caching signatures. The explanations below will still be largely accurate throughout the rollout.
+
+To achieve the “one click” gem signing experience, the RubyGems client orchestrates a multi-step flow with several services. The design of this flow means that signing keys are ephemeral and that signatures are published to a trusted public transparency log. First we will give background about ephemeral keys and transparency logs, then we will describe the gem signing and gem signature verification flows.
+
+### Background concepts
+
+#### Ephemeral signing keys
+
+Historically, digital signature schemes have been based on dual-key encryption. The signer possesses a _private signing key_, which is not shared with anyone else. This key is used to sign artifacts, creating _signatures_. The signer then publishes a _public certificate_, which includes a _public verification key_. The public certificate binds information about the signer’s identity together with the public verification key into a single unit. The public certificate can be distributed widely. Verifiers can use the public certificate to verify that the private signing key was used to create a signature for a given artifact. In gem signing today, most certificates are self-signed, meaning that there is no independent root of trust for the claims of identity made in 
```

Comment:
A colleague made this suggestion:
> I never heard "dual-key encryption" before - the description that comes after matches what's commonly known as "public-key cryptography" so I wanted to suggest to use that industry standard term

Makes sense to me!

Reply:
I could've sworn I'd seen it before, but "dual-key" seems to be a _very_ niche term for what it is. I think "public-key cryptography" would be better in this case.